File: zkstrap/web/js/org/effortless/zkstrap/Skin.js
org.effortless.zkstrap.Skin = zk.$extends(zk.Widget, {
  _name : '', // default value

  getName : function() {
    return this._name;
  },

  setName : function(newValue) {
    if (this._name != newValue) {
      this._name = newValue;
//      console.log('HLEOOO!2');
//       	this.parent.parent.setSclass(this._name);
//       	this.parent.parent.setSclass('skin-blue');
//        document.body.addClass(this._name);
//        document.body.addClass('skin-blue');
        document.body.className = this._name;
//      console.log('HLEOOO!3');
//      if (this.desktop) {
//      }
    }
  }
  
});

File: zkstrap/classes/org/effortless/zkstrap/Skin.java
package org.effortless.zkstrap;

public class Skin extends org.zkoss.zk.ui.HtmlBasedComponent {

	public Skin () {
		super();
		initiate();
	}
	
	protected void initiate () {
//		this.setZclass("label label-default");
////		this.setSclass("default");
	}
	
	protected String name = ""; // a data member

	public String getName() {
		return this.name;
	}
	
	public void setName(String newValue) {
		if (!this.name.equals(newValue)) {
			this.name = newValue;
			smartUpdate("name", this.name);
		}
	}
	
	protected void renderProperties(org.zkoss.zk.ui.sys.ContentRenderer renderer) throws java.io.IOException {
		super.renderProperties(renderer);
		render(renderer, "name", this.name);
	}

//	public void setBclass (String newValue) {
//		setZclass("label label-" + newValue);
//	}
	
//	public void setSclass (String newValue) {
//		newValue = "label-" + newValue;
//		super.setSclass(newValue);
//	}
	
}
